src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -119,7 +124,21 @@ export const SpawnHostModal: React.VFC<SpawnHostModalProps> = ({
     },
     refetchQueries: ["MyHosts", "MyVolumes", "GetMyPublicKeys"],
```

Comment:
One thing i noticed is that all the queries for this modal are loaded on page load despite the spawn host modal never being opened. We can avoid this using the [`skip`](https://www.apollographql.com/docs/react/api/react/hooks#skip) command on the queries.  Particularly GET_DISTROS, GET_MY_VOLUMES, GET_MY_PUBLIC_KEYS,

Reply:
Agreed, another approach I use sometimes is conditionally rendering the modal based on the open state in the parent component. E.g.

```js
{open && <Modal {...props} />
```

This has the added benefit of resetting any form states upon closing the modal (which is generally expected behavior for modals IMO) since the component is removed from the DOM.